src/common/browser-adapters/browser-adapter-factory.ts

```diff
@@ -1,18 +1,41 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+import { BrowserEventManager } from 'common/browser-adapters/browser-event-manager';
 import { ChromiumAdapter } from 'common/browser-adapters/chromium-adapter';
 import { FirefoxAdapter } from 'common/browser-adapters/firefox-adapter';
 import { WebExtensionBrowserAdapter } from 'common/browser-adapters/webextension-browser-adapter';
+import { DictionaryStringTo } from 'types/common-types';
 import * as UAParser from 'ua-parser-js';
+import browser, { Events } from 'webextension-polyfill';
 
 export class BrowserAdapterFactory {
     public constructor(private readonly uaParser: UAParser) {}
 
-    public makeFromUserAgent(): WebExtensionBrowserAdapter {
+    public makeFromUserAgent(browserEventManager: BrowserEventManager): WebExtensionBrowserAdapter {
+        const browserEvents = this.getDefaultBrowserEvents();
+
         if (this.uaParser.getEngine().name === 'Gecko') {
-            return new FirefoxAdapter();
+            return new FirefoxAdapter(browserEventManager, browserEvents);
         } else {
-            return new ChromiumAdapter();
+            return new ChromiumAdapter(browserEventManager, browserEvents);
         }
     }
+
+    protected getDefaultBrowserEvents(): DictionaryStringTo<Events.Event<any>> {
+        const browserEvents: DictionaryStringTo<Events.Event<any>> = {
```

Comment:
This specific list is tightly coupled to the `webextension-browser-adapter` implementation referencing the keys, and it isn't used anywhere else, so I think it would make more sense for this to live there.